woocommerce/woocommerce-gutenberg-products-block
# Introduce ContainerQueryContextProvider - hide order summary on mobile

Introduces a new context provider named `ContainerQueryContextProvider`. This consumes `useContainerQueries` so that all children in the tree can see the container size, for example, to know if isMobile is true or not.

This is used on the checkout order summary area so the default state of that is collapsed unless we're on a larger width device/container.

Fixes #2423

### How to test the changes in this Pull Request:

1. On desktop, view the checkout. Order summary is open.
2. Switch to mobile view. View checkout. Order summary is closed.
3. Check is-x classes remain on the checkout and cart wrappers.
